service.go

```diff
@@ -188,6 +188,7 @@ func (s *Service) createHTTPComponent() (Component, error) {
 	log.Infof("creating default HTTP component at port %s", port)
 
 	b := http.NewBuilder()
+	b.WithPort(int(portVal))
```

Comment:
b := http.NewBuilder().WithPort(port)

Reply:
`port` is a string, and `portVal` is an int64. I think it's quite concise as is; should we add an intermediate step to convert to int beforehand?

Reply:
Done, sorry.